torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -290,13 +290,32 @@ auto handle_torch_function_no_python_arg_parser(
   py::object ret;
   py::object mode_obj;
 
+  // Of our overloaded types, filter out the FakeTensor arguments.
+  // FakeTensor is special, and gets lower precdence in the dispatcher ordering.
+  std::vector<py::handle> overloaded_non_fake_args;
+  std::vector<py::handle> overloaded_fake_args;
+  for (auto& arg : overloaded_args) {
+    auto curr_type = get_type_of_overloaded_arg(arg.ptr());
+    auto curr_type_name = PyObject_GetAttrString(curr_type, "__name__");
+    auto curr_type_str = std::string(PyUnicode_AsUTF8(curr_type_name));
+    if (curr_type_str != "FakeTensor") {
```

Comment:
nuh uh. I'm not sure whether or not overall this makes sense, but if you're going to do this, get your hands on the FakeTensor type object (by doing a py::import thing) and then do a regular isinstance test